lib/pbench/server/cache_manager.py

```diff
@@ -274,59 +277,51 @@ def cache_map(self, tar_path: Path):
             for l_path in dir_list:
                 tar_info = FileInfo(l_path)
                 curr[l_path.name] = {"details": tar_info}
-                # Note: have to handle symlink case as it
-                # is not being checked whether it is a symlink,
-                # it is checking for the redirected path. Currently,
-                # as I have used file for the symlink it is not added it
-                # the queue, but if it a symlink is pointing to a directory
-                # it turns out to be an endless loop. There may be case
-                # that the way I am creating symlink in tests is wrong because
-                # when I am testing manually I am not seeing the same case.
                 if l_path.is_dir():
                     dir_queue.append((l_path, curr))
         self.cachemap = cmap
 
     @staticmethod
-    def traverse_cmap(path, cachemap):
+    def traverse_cmap(path: Path, cachemap: dict):
+        name = path.name
         files_list = path.parts
         c_map = cachemap
 
         for file_l in files_list:
-            if file_l != ["/", "//"]:
+            if file_l not in ("/", "//"):
                 c_map = c_map[file_l]
                 if c_map["details"].type == "directory":
-                    if "children" in c_map:
+                    if "children" in c_map and file_l != name:
                         c_map = c_map["children"]
-                elif c_map["details"].type == "symlink":
-                    resolve_path = c_map["details"].resolve_path
-                    c_map = Tarball.traverse_cmap(resolve_path, cachemap)
 
         return c_map
 
-    def get_info(self, path):
-        c_map = Tarball.traverse_cmap(path, self.cachemap)
+    def get_info(self, path: Path):
+        # FixMe: Handle a Symlink
+        c_map = self.traverse_cmap(path, self.cachemap)
         fd_info = {}
-
-        if "details" in c_map:
-            if c_map["details"].type == "file":
-                fd_info = {
-                    "location": c_map["details"].location,
-                    "name": c_map["details"].name,
-                    "size": "1024",
-                    "type": c_map["details"].type,
-                }
-            elif c_map["details"].type == "directory":
-                fd_info = {
-                    "location": c_map["details"].location,
-                    "name": c_map["details"].name,
-                    "type": c_map["details"].type,
-                }
-        else:
-            for key, value in c_map.items():
+        children = c_map["children"] if "children" in c_map else False
```

Comment:
it is a dict.

Reply:
If we want, we can use a "real object" which is derived from `dict` (or which otherwise provides the same interface, or which provides whatever interface the "jsonify" function requires...) -- in that way we get the advantages of having "types" while still being able to convert it to JSON in a straightforward way.

Reply:
Yeah, that's what I "started to attempt" with `JSONMap` in the client library, although it's far from perfect and wasn't intended to allow for modification. For example, `(o = JSONMap({...}); o["key"] = value` should dynamically add `o.json["key"]` while also defining the `o.key` attribute.)

However this discards many of the advantages of real objects, in that `o["foobar"]` won't fail if `foobar` isn't supposed to be an attribute, while a real class will at least generate an IDE complaint.